### DISCUSSION TITLE
Does Slint suit for large desktop application?

### DISCUSSION THREAD

**Discussion by asuper0** [slint-ui/slint#6628] [General] [answered]
A large desktop application may has a lot of pages, work with some huge database, and table widget with thousands of data, e.g., ERP system. As far as I see, Slint load everything defined in `.slint`, even if the user has not enter the page yet ( i.e. it's not lazy). So do you think Slint is suit for it?

**Comment by tronical** [comment, marked as answer]
If you use [conditional elements](https://docs.slint.dev/latest/docs/slint/src/language/syntax/conditions#conditional-element) then whatever is declared inside is only created, thus memory allocated, when the condition is true.

**Reply by bjorn** [reply]
Also, due to [`ListView`](https://docs.slint.dev/latest/docs/slint/src/language/widgets/listview) only instantiating elements that are visible, models with thousands of rows are not a problem either.

The only issue for "large applications" I've encountered is the compile time. For my app [Raccoin](https://raccoin.org/), compile is taking almost 1 minute after changing any `.slint` file in release mode. There have been improvements to this in recent Slint releases it seems though, since it is only taking 11 seconds in debug builds now.
